File: 001/pseudoinverse.py
import time
import numpy as np

points = [[0,0],[1,1],[2,4],[3,9]]


avg = []
for i in range(0,10):
    temp = time.time()
    x = np.array([[1,0],[1,1],[1,2],[1,3]])
    x_inverse = np.linalg.pinv(x)
    y = np.array([[0],[1],[4],[9]])
    beta = np.dot(x_inverse, y)
    avg.append(time.time() - temp)

print('python:', sum(avg) / 10)

File: 001/pseudoinverse.jl
using LinearAlgebra

avg = []
for i = 1:10
    start_time = time_ns()
    x = [1 0; 1 1; 1 2; 1 3;]
    x_inverse = pinv(x)
    y = [0; 1; 4; 9]
    beta = x_inverse * y
    append!(avg, (time_ns() - start_time))
end

println("julia: ", sum(avg) / 1000000000 / 10)